src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -1088,16 +1088,35 @@ void Deoptimization::reassign_fields(frame* fr, RegisterMap* reg_map, GrowableAr
     Klass* k = java_lang_Class::as_Klass(sv->klass()->as_ConstantOopReadValue()->value()());
     Handle obj = sv->value();
     assert(obj.not_null() || realloc_failures, "reallocation was missed");
+#ifndef PRODUCT
```

Comment:
Why remove the def?
Only support debug info printing in debug mode?

Reply:
> Why remove the def? Only support debug info printing in debug mode?

Yes. I checked that the "PrintDeoptimizationDetails" flag is a develop flag are settable / visible only during development and are constant in the PRODUCT version.